### PR TITLE
K8s 1.25 for local clusters

### PR DESCRIPTION
**Pull request by git-ival**
- K8s 1.25.x support for local clusters
- Update READMEs
- Move rancher-specific security group terraform in `aws-infra/infra.tf` to new file and make it conditional
- Enable `rancher2_settings` to be set via input vars
- Add optional method to input additional user data script parts
- Add various validations to complex input variables

Adds support for the 1st entry of this task rancher/qa-tasks#617
Closes rancher/qa-tasks#628

### CODE REVIEW

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf [hunk 1]
```diff
@@ -1,79 +1,79 @@
 ### install common components for Rancher local clusters (cert-manager + Rancher)
-module "install_common" {
-  source = "../../../install-common"
-  providers = {
-    rancher2 = rancher2.bootstrap
-  }
+# module "install_common" {
```
Comment (floatingman) [inline rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:2]: Is this going to be used in the future?

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf [hunk 2]
```diff
@@ -1,79 +1,79 @@
...
 
-  kube_config_path = var.kube_config_path
+#   kube_config_path = var.kube_config_path
 
-  subdomain           = local.rancher_subdomain
-  domain              = local.rancher_domain
-  install_certmanager = true
-  install_rancher     = true
-  certmanager_version = "1.8.1"
+#   subdomain           = local.rancher_subdomain
```
Comment (floatingman) [inline rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:11]: Is this going to be used in the future?

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf [hunk 3]
```diff
@@ -1,79 +1,79 @@
...
-  provider = rancher2.admin
+# data "rancher2_cluster" "local" {
+#   provider = rancher2.admin
 
-  name = "local"
-  depends_on = [
-    module.install_common
-  ]
-}
+#   name = "local"
```
Comment (floatingman) [inline rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:30]: Is this going to be used in the future?

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf [hunk 4]
```diff
@@ -1,79 +1,79 @@
...
-  use_v2        = false
-  rancher_url   = module.install_common.rancher_url
-  rancher_token = module.install_common.rancher_token
-  charts_branch = "release-v2.5"
-  chart_version = "16.6.1+up16.6.0"
-  values        = "../../../charts/files/rancher_monitoring_chart_values.yaml"
-  cluster_id    = data.rancher2_cluster.local.id
-  project_id    = data.rancher2_cluster.local.default_project_id
-}
+#   use_v2        = false
```
Comment (floatingman) [inline rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:43]: Is this going to be used in the future?

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf [hunk 5]
```diff
@@ -1,79 +1,79 @@
...
 module "controller_metrics" {
   source          = "../../../rancher-controller-metrics"
-  rancher_token   = module.install_common.rancher_token
+  rancher_token   = var.rancher_token
   rancher_version = var.rancher_version
 
-  depends_on = [
-    module.rancher_monitoring
-  ]
+  # depends_on = [
```
Comment (floatingman) [inline rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:59]: Is this going to be used in the future?

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf [hunk 6]
```diff
@@ -1,79 +1,79 @@
...
-  create_new  = true
-  annotations = { example = "annotation" }
-  labels      = { example = "label" }
-  description = "Example description of secret"
-  project_id  = data.rancher2_cluster.local.default_project_id
-  name        = "tf-example-secret"
-  namespace   = "default"
-  data        = { example = "my data value" }
-}
+#   use_v2      = false
```
Comment (floatingman) [inline rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:70]: Is this going to be used in the future?

**Review by floatingman — CHANGES_REQUESTED** [review]
Just a few comments

File: cluster-with-nodes/modules/downstream-k3s-nodes/files/worker_userdata.tmpl [hunk 7]
```diff
@@ -1,5 +1,5 @@
 #cloud-config
-ssh_authorized_keys: 
+ssh_authorized_keys:
   - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDaitXW9CBM/yxEhVaBQ9WEB8KEdwbzlMk6M2URfbr07mA9Rk3ssWf2hS3RMwF76rasRLmrjsge1Q9cc8uck722vs8qCpwz/aMnsHF8kfnifkv3JKgfYCTkVzbsW7HzNmmkbsBVPS2tlVDpG+cHQNwZMgo1MGVJqAvPnrKlLzSj/XVeBKYV8a7/GLwrJTiXegitkyE3i1L42aMwzGDwEDnKkFBYW1JXXim0j97ztWy8YaScuptqd9WJ5NfHgUGfhFSYKlEX+FP5x0oyLVgGRmoUh7NellALQ9mWW+tJgSd7oWPoeJtHLAVMoiX02w1OtBUOFq1s0IJ9nDdU/I9v+grP chriskim@Endeavor
```
[the remark this answers is not in the document]
Comment (git-ival) [reply cluster-with-nodes/modules/downstream-k3s-nodes/files/worker_userdata.tmpl:3]: This was here before, just a formatting change. Also this module isnt currently used.

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf (hunk 2 above)
Comment (git-ival) [reply rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:11]: Will remove.

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf [hunk 8]
```diff
@@ -1,79 +1,79 @@
...
-  helm_rancher_chart_values_path = "../../../install-common/files/rancher_chart_values.tftpl"
-  letsencrypt_email              = var.letsencrypt_email
-  rancher_image                  = "rancher/rancher"
-  rancher_version                = var.rancher_version
-  rancher_password               = var.rancher_password
-  use_new_bootstrap              = false
-  rancher_node_count             = var.rancher_node_count
-  cattle_prometheus_metrics      = true
-}
+#   helm_rancher_chart_values_path = "../../../install-common/files/rancher_chart_values.tftpl"
```
[the remark this answers is not in the document]
Comment (git-ival) [reply rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:17]: Will remove.

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf (hunk 3 above)
Comment (git-ival) [reply rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:30]: Will remove.

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf (hunk 4 above)
Comment (git-ival) [reply rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:43]: Will remove.

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf (hunk 5 above)
Comment (git-ival) [reply rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:59]: Will remove.

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf [hunk 9]
```diff
@@ -1,79 +1,79 @@
...
+  #   module.rancher_monitoring
+  # ]
 }
 
-module "secret" {
-  source = "../../../rancher-secret"
-  providers = {
-    rancher2 = rancher2.admin
-  }
+# module "secret" {
```
[the remark this answers is not in the document]
Comment (git-ival) [reply rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:64]: Will remove.

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf (hunk 6 above)
Comment (git-ival) [reply rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:70]: Will remove.

File: rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf (hunk 1 above)
Comment (git-ival) [reply rancher-cluster-operations/rancher-examples/rancher-setup/2.5.x/main.tf:2]: Will remove.

**Review by floatingman — APPROVED** [review]
Looks good.

**Pull request merged**